bin/cheribsdtest/Makefile.cheribsdtest

```diff
@@ -46,39 +46,7 @@ SRCS+=	cheribsdtest_cheriabi_open.c
 #SRCS+=	cheribsdtest_vm_swap.c
 #CFLAGS.cheribsdtest_vm_swap.c+=-Wno-cheri-capability-misuse
 
-.if ${MACHINE} == mips
-.PATH: ${SRCTOP}/sys/mips/cheri
-SRCS+=	cheri_memcpy_c.S
-CFLAGS.cheri_memcpy_c.S=-D__KERN_FUNC_PREFIX -D_KERNEL -DINVARIANTS
-CFLAGS+=	-DKERNEL_MEMCPY_TESTS
-.endif
-
-.if ${MACHINE} == "mips"
-.ifdef CHERI_C_TESTS
-CHERI_C_TESTS_DIR=	${SRCTOP}/contrib/subrepo-cheri-c-tests
-.if exists(${CHERI_C_TESTS_DIR}/Makefile)
-.PATH: ${CHERI_C_TESTS_DIR} \
-		${CHERI_C_TESTS_DIR}/clang-hybrid \
-		${CHERI_C_TESTS_DIR}/clang-purecap \
-		${CHERI_C_TESTS_DIR}/libc
-CFLAGS+=	-DCHERI_C_TESTS \
-		-I${CHERI_C_TESTS_DIR}
-
-CFLAGS+=	-DTEST_CUSTOM_FRAMEWORK -I${CHERIBSDTEST_DIR} \
-		-DHAVE_MALLOC_USUABLE_SIZE
-TEST_SRCS!=	grep ^DECLARE_TEST ${CHERI_C_TESTS_DIR}/cheri_c_testdecls.h | \
-		    sed -e 's/.*(\([^,]*\),.*/\1.c/'
-SRCS+=	test_runtime.c	\
-	${TEST_SRCS}
-
-.for test in ${TEST_SRCS}
-# Avoid the need for extern declarations for variables/functions that are
-# declared global so that the compiler emits them
-CFLAGS.${test}+=-Wno-missing-prototypes -Wno-missing-variable-declarations
-.endfor
-.endif
-.endif
-.endif
+.include "${CHERIBSDTEST_DIR}/${MACHINE}/Makefile.cheribsdtest"
```

Comment:
Could maybe use '.sinclude' and then you don't need empty files for architectures without MD tests?

Another option might be to use <bsd.arch.inc.mk> and then the Makefile.<arch> lives in cheribsdtest directly (and are optional)

Reply:
I thought about that but I didn't want it to silently stop including tests if something broke